tidy3d/components/medium.py

```diff
@@ -71,6 +74,75 @@ def _eps_model(self, frequency: float) -> complex:
 """ Medium Definitions """
 
 
+class AbstractNonlinearSpec(ABC, Tidy3dBaseModel):
+    """Abstract specification for adding a nonlinearity to a medium.
+
+     Note
+    ----
+    The nonlinear constitutive relation is solved iteratively; it may not converge
+    for strong nonlinearities. Increasing `numiters` can help with convergence.
+    """
+
+    numiters: pd.PositiveInt = pd.Field(
+        1,
+        title="Number of iterations",
+        description="Number of iterations for solving nonlinear constitutive relation.",
+    )
+
+    @pd.validator("numiters", always=True)
+    def _validate_numiters(cls, val):
+        """Check that numiters is not too large."""
+        if val > NONLINEAR_MAX_NUMITERS:
+            raise ValidationError(
+                "NonlinearSpec.numiters must be less than "
```

Comment:
let's put anything code related in single quotes `'`. `'NonlinearSpec.numiters'`.